src/core/device/platform.cc

```diff
@@ -120,6 +121,26 @@ Platform::CreateCudaGPUs(const size_t num_devices, size_t init_size) {
 
 const vector<shared_ptr<Device>>
 Platform::CreateCudaGPUsOn(const vector<int> &devices, size_t init_size) {
+
```

Comment:
it would be better to add a lock here.
if two threads call this function at the same time, this function's output is then not deterministic.